main.py

```diff
@@ -298,27 +302,33 @@ def generate_query(selector, ip_translator, mlab_site_resolver):
   start_time_datetime = selector.start_time
   end_time_datetime = start_time_datetime + datetime.timedelta(seconds = selector.duration)
 
-  network_lookup_found_blocks = ip_translator.find_ip_blocks(
-      selector.client_provider)
-  if len(network_lookup_found_blocks) == 0:
-    raise NoClientNetworkBlocksFound(selector.client_provider)
-
-  server_ips = []
-  try:
-    for retrieved_site_ip in mlab_site_resolver.get_site_ips(selector.site,
-                                                             mlab_project = selector.mlab_project):
-      server_ips.append(retrieved_site_ip)
-      logger.debug("Found IP for {site} of {site_ip} on test {test}.".format(
-          site=selector.site, site_ip = retrieved_site_ip, test = selector.mlab_project))
-  except Exception as caught_error:
-    raise MLabServerResolutionFailed(caught_error)
-
+  if selector.client_provider != None:
+    client_ips = ip_translator.find_ip_blocks(selector.client_provider)
+    if len(client_ips) == 0:
+        raise NoClientNetworkBlocksFound(selector.client_provider)
+  else:
+    client_ips = None
```

Comment:
For `client_ips` and `server_ips` we're currently making it either `None` or a non-empty list. Can we simplify the code by making these always be lists and they're either 0 or non-empty. I think we can avoid explicit checks for `None` before each operation and make it so that the loops just do nothing when the list is empty. i.e., it would be like:

```
client_ips = []
if selector.client_provider != None:
    client_ips = ip_translator.find_ip_blocks(selector.client_provider)
    if not client_ips:
        raise NoClientNetworkBlocksFound(selector.client_provider)
```